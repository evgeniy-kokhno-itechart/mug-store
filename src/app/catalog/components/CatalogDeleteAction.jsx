import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Modal } from '../../shared';

class CatalogDeleteAction extends Component {
  handleDeleteAction = () => {
    const { productId, handleAction } = this.props;
    handleAction(productId);
  };

  render() {
    const { productTitle } = this.props;
    return (
      <Modal
        id="product-deletion-confirmation"
        buttonLabel="Delete"
        buttonClasses="btn btn-danger btn-sm"
        title="Confirm product deletion"
        text={`You are about to completely delete ${productTitle} from the database!`}
        textConfirm="Confirm"
        textAbort="Dismiss"
        onConfirm={this.handleDeleteAction}
      />
    );
  }
}

CatalogDeleteAction.propTypes = {
  productId: PropTypes.string.isRequired,
  productTitle: PropTypes.string.isRequired,
  handleAction: PropTypes.func.isRequired,
};

export default CatalogDeleteAction;
