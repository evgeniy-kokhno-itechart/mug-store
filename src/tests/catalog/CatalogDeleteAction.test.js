/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import CatalogDeleteAction from '../../app/catalog/components/CatalogDeleteAction';

configure({ adapter: new Adapter() });

describe('<CatalogDeleteAction />', () => {
  let CatalogDeleteActionWrapper;

  beforeEach(() => {
    const handleAction = jest.fn();
    CatalogDeleteActionWrapper = shallow(<CatalogDeleteAction productId="1" productTitle="testProduct" handleAction={handleAction} />);
  });

  it('check render properly', () => {
    expect(CatalogDeleteActionWrapper).toMatchSnapshot();
  });
});
