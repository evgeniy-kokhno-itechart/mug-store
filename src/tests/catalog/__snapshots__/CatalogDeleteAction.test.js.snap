// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatalogDeleteAction /> check render properly 1`] = `
<Modal
  buttonClasses="btn btn-danger btn-sm"
  buttonLabel="Delete"
  id="product-deletion-confirmation"
  onConfirm={[Function]}
  text="You are about to completely delete testProduct from the database!"
  textAbort="Dismiss"
  textConfirm="Confirm"
  title="Confirm product deletion"
/>
`;
